src/sagas/nanoContract.js

```diff
@@ -106,8 +117,143 @@ export function* registerNanoContract({ payload }) {
   yield put(nanoContractRegisterSuccess({ entryKey: ncEntryKey, entryValue: ncEntryValue }));
 }
 
+/**
+ * @typedef {Object} RawNcTxHistory
+ * @property {string} hash
+ * @property {Object[]} inputs
+ * @property {Object[]} outputs
+ * @property {string} nc_args
+ * @property {string} nc_id
+ * @property {string} nc_method
+ * @property {string} nc_pubkey
+ * @property {number} nonce
+ * @property {Object[]} parents
+ * @property {number} signal_bits
+ * @property {Date} timestamp
+ * @property {string[]} tokens
+ * @property {number} version
+ * @property {number} weight
+ *
+ * @typedef {Object} RawNcTxHistoryResponse
+ * @property {boolean} success
+ * @property {number|null} after
+ * @property {RawNcTxHistory} history
+ */
+
+/**
+ * @typedef {Object} NcTxHistory - Nano Contract's Transactions History
+ * @property {string} txId - transaction hash
+ * @property {number} timestamp
+ * @property {string} tokens - list of token's uid
+ * @property {{[uid: string]: Object}} balance - balance per token
+ * @property {boolean} isVoided - flag for transaction void
+ * @property {string} ncId - ID of Nano Contract
+ * @property {string} ncMethod - method called on Nano Contract
+ * @property {string} blueprintId - id of the Blueprint instantiated by Nano Contract
+ * @property {string} caller - address of the caller
+ * @property {'mine'|'nc'|'oracle'|'wallet'} callerOrigin - caller's origin
+ */
+
+/**
+ * @param {RawNcTxHistory} each
+ * @returns {NcTxHistory}
+ */
+function toNcTxHistory(each) {
+  return {
+    txId: each.hash,
+    timestamp: each.timestamp,
+    tokens: each.tokens,
+    ncId: each.nc_id,
+    ncMethod: each.nc_method,
+    // balance: getTxBalance.bind(transactionUtils)(each),
```

Comment:
I prepared a balance method but I realize that it would need to download the input's transaction. I'm figuring out a solution yet.